File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

char alpabet[] = {'A', 'E', 'I', 'O', 'U'};
vector<string> list;

void DFS(string currentWords)
{
    if(currentWords.length() > 5) return;
    
    list.push_back(currentWords);
    
    for(int i = 0; i < sizeof(alpabet); i++)
        DFS(currentWords + alpabet[i]);
    
    return;
}

int solution(string word) {
    DFS("");
    
    for(int i = 0 ; i < list.size(); i++)
    {
        if(list[i] == word) return i;
    }
    return 0;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cs
using System;
using System.Collections.Generic;

public class Solution {
    public char[] alphabet = {'A', 'E', 'I', 'O', 'U'};
    public List<string> list = new List<string>();
    
    public void DFS(string currentWord, int len)
    {
        if(len > 5) return;
        
        list.Add(currentWord);
        
        for(int i = 0; i < alphabet.Length; i++)
            DFS(currentWord + alphabet[i], len + 1);
        
        return;
    }
    
    public int solution(string word) {
        DFS("", 0);
        return list.IndexOf(word);
    }
}